### ISSUE TITLE
branches.only filter not supported on travis-ci.org

### ISSUE THREAD

**Issue by davethegr8** [travis-ci/travis-ci#5813]
According to https://docs.travis-ci.com/user/customizing-the-build/ I should be able to limit branches that travis builds by using the following in my travis.yml:

```
branches:
  only:
    - master
```

However, his doesn't seem to be supported for travis-ci.org. My project https://github.com/cascadiajs/2016.cascadiajs.com is using travis to automatically deploy our website. Last night, I opened a PR https://github.com/cascadiajs/2016.cascadiajs.com/pull/5 to add staff information to our homepage. Despite this being in a branch (staff-preview) and not merged into master, travis built anyway and announced the staff information before we were ready. 

Is this an expected limitation of travis-ci.org? Am I doing something wrong?

**Comment by BanzaiMan** [comment]
Pull Requests do not trigger a build. https://docs.travis-ci.com/user/deployment/#Pull-Requests

Once the PR is merged, it'll be deployed.

**Comment by BanzaiMan** [comment]
Actually, you are not using our deployment scheme (https://github.com/cascadiajs/2016.cascadiajs.com/blob/2e760f947061c7cbb055ded6cfbb9f5c945fa743/.travis.yml) so the comment above does not apply.

```yaml
branches:
  only:
    - master
```

in your case only applies to whether or not the build runs, and it did for https://travis-ci.org/cascadiajs/2016.cascadiajs.com/builds/117890293. The project is using a custom `after_success` step `npm run deploy`, which says it succeeded (https://travis-ci.org/cascadiajs/2016.cascadiajs.com/builds/117890293#L398). You should check what the process is doing.

You should note, though, that *any* pull request can run this code. This can have bad consequences for your site.

**Comment by davethegr8** [comment]
Ah, thanks I think I understand now. Much appreciated :smile:
